R-packages/covidcast/tests/testthat/test-covidcast.R

```diff
@@ -0,0 +1,26 @@
+library(covidcast)
+library(mockery)
+
+test_that("covidcast_meta", {
+  stub(covidcast_meta, ".request",
+       list(message = "success",
+            epidata = data.frame(
+              min_time = c("20200101", "20201002"),
+              max_time = c("20200102", "20201003"),
+              max_issue = c("20200404", "20201101"))))
+
+  expect_equal(covidcast_meta(),
+               structure(
+                 data.frame(
+                   min_time = as.Date(c("2020-01-01", "2020-10-02")),
+                   max_time = as.Date(c("2020-01-02", "2020-10-03")),
+                   max_issue = as.Date(c("2020-04-04", "2020-11-01"))
+                 ),
+                 class = c("covidcast_meta", "data.frame")
+               ))
+
+  stub(covidcast_meta, ".request",
+       list(message = "argle-bargle"))
+
+  expect_error(covidcast_meta())
```

Comment:
Does this allow specification of which error is expected?

Reply:
Not if `covidcast_meta()` uses `stop()`, which does not specify a specific error type -- unless you regexp-match against the error message.

I've just changed `covidcast_meta()` to use `abort()` with a specific error class, so we can catch and handle it, and the test can verify it was raised correctly.